contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1286,8 +1304,19 @@ create_bbf_authid_user_ext(CreateRoleStmt *stmt, bool has_schema, bool has_login
 
 	if (has_login)
 	{
+		char *db_name = get_cur_db_name();
+		int         save_sec_context;
+		Oid         save_userid;
+
 		verify_login_for_bbf_authid_user_ext(login);
 		login_name_str = login->rolename;
+		/* Revoke guest user from login as login now has a mapped user in current database. */
+		GetUserIdAndSecContext(&save_userid, &save_sec_context);
+		SetUserIdAndSecContext(get_sa_role_oid(), save_sec_context | SECURITY_LOCAL_USERID_CHANGE);
```

Comment:
Can you please add a comment explaining - Why we are using SA and not bbf_role_admin?